frontend/pages/views/PhotographerListView.js

```diff
@@ -3,66 +3,273 @@ import Footer from "../../components/Footer";
     hrefFunc(number) {
         return `/photographer/${number}`;
-    };
+    }
 
     srcFunc(number) {
         return `${this.props.photoListDir}/${number}_photo.jpg`;
-    };
+    }
 
     getPhotoList() {
         const photoSize = [100, 100];
         return this.state.photographers.map((photog, k) => {
             return (
                 <li className="col-2 col-lg-2 one-photographer list-inline-item" key={k}>
                     <div className="child">
-                        <a key={k}
-                           href={this.hrefFunc(photog.number)}>
+                        <a key={k} href={this.hrefFunc(photog.number)}>
                             <img
                                 alt={photog.number}
                                 width={photoSize[0]}
-                                src={this.srcFunc(photog.number)}/>
+                                src={this.srcFunc(photog.number)}
+                            />
                         </a>
                         <p>{photog.name ? photog.name : "No Name"}</p>
                     </div>
                 </li>
             );
         });
+    }
 
+    handleScroll = () => {
+        // Detecting scroll end adapted from https://stackoverflow.com/a/4620986
+        if (this.scrollOverTimer !== null) {
+            clearTimeout(this.scrollOverTimer);
+        }
+
+        let dropdowns = document.getElementsByClassName("dropdown-items");
+        for (const elt of dropdowns) {
+            elt.classList.add("d-none");
+        }
+
+        this.scrollOverTimer = setTimeout(() => {
+            // If the timer elapses, show to
+            // this.setState({retract:true});
+            document.getElementsByClassName("banner")[0].classList.add("grow");
+            document.getElementsByClassName("banner")[0].classList.remove("shrink");
+        }, 1000);
+
+        document.getElementsByClassName("banner")[0].classList.add("shrink");
+        document.getElementsByClassName("banner")[0].classList.remove("grow");
+
+        if (window.scrollY > 70) {
+            document.getElementsByClassName("banner")[0].style.position = "fixed";
+            document.getElementsByClassName("banner")[0].style.top = "0px";
+        } else {
+            document.getElementsByClassName("banner")[0].style.position = "absolute";
+            document.getElementsByClassName("banner")[0].style.top = "70px";
+        }
+    };
+
+    toggleActiveDropdown = (dropDown) => {
+        this.setState({activeDropdown: dropDown});
+    };
+
+    componentDidMount() {
+        // if (!this.galleryRef) {
+        //     return;
+        //   }
+        window.addEventListener("scroll", this.handleScroll);
+    }
+
+    componentDidUpdate(prevState) {
+        console.log(this.state.activeDropdown);
+
+        if (prevState.activeDropdown !== this.state.activeDropdown) {
+            if (this.state.activeDropdown) {
+                document.getElementsByClassName("overlay")[0].classList.add("show");
```

Comment:
same deal as above - rather than querying repeatedly, query and save the element in a var.